Docs/reference/content/reference/driver/crud/reading.md

```diff
@@ -312,4 +312,224 @@ await OutAsync("myNewCollection");
 ```
 ```json
 { $out: 'myNewCollection' }
-```
\ No newline at end of file
+```
+
+### Text Search
+[Atlas Search](https://docs.atlas.mongodb.com/atlas-search) makes it easy to build fast, relevance-based search capabilities on top of your MongoDB data. Try it today on [MongoDB Atlas](https://www.mongodb.com/cloud/atlas), our fully managed database as a service.
+
+The following examples can be used alongside the [Atlas Search Tutorial](https://docs.atlas.mongodb.com/reference/atlas-search/tutorial/)
+
+```csharp
+var collection = client.GetDatabase("sample_mflix").GetCollection<BsonDocument>("movies");
+var searchStage = @"
+{
+	$search: {
+    	'text': {
+    	    'query': 'baseball',
+            'path': 'plot'
+        }
+    }
+}";
+var projection = @"
+{
+    _id : 0,
+    title: 1,
+    plot: 1
+}";
+
+var pipeline = new EmptyPipelineDefinition<BsonDocument>()
+    .AppendStage<BsonDocument, BsonDocument, BsonDocument>(searchStage)
+    .Limit(5)
+    .Project(projection);
+
+var moviesAboutBaseball = collection.Aggregate(pipeline).ToList();
+Console.WriteLine("Movies about baseball:");
+foreach (var movie in moviesAboutBaseball)
+{
+    Console.WriteLine($"  {movie}");
+}
+string complexSearch = @"
+{
+  $search: {
+    'compound': {
+      'must': [ {
+        'text': {
+           'query': ['Hawaii', 'Alaska'],
+           'path': 'plot'
+        },
+      },
+      {
+        'regex': {
+           'query': '([0-9]{4})',
+           'path': 'plot',
+           'allowAnalyzedField': true
+        }
+      } ],
+      'mustNot': [ {
+        'text': {
+          'query': ['Comedy', 'Romance'],
+          'path': 'genres'
+        }
+      },
+      {
+        'term': {
+          'query': ['Beach', 'Snow'],
+          'path': 'title'
+        }
+      } ]
+    }
+  }
+}";
+
+var projectionWithGenre = @"
+{
+    'title': 1,
+    'plot': 1,
+    'genres': 1,
+    '_id': 0
+}";
+
+
+var complexPipeline = new EmptyPipelineDefinition<BsonDocument>()
+    .AppendStage<BsonDocument, BsonDocument, BsonDocument>(complexSearch)
+    .Project(projectionWithGenre);
+
+var complexQueryResults = collection.Aggregate<BsonDocument>(complexPipeline).ToList();
+Console.WriteLine("Complex query results: ");
+foreach (var movie in complexQueryResults)
+{
+    Console.WriteLine($"  {movie}");
+}
+var queryUsingKeywordAnalyzer = @"
```

Comment:
This is a new example right? We should have some white space and probably some text separating it from the previous example.